rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/InstallPackageCommand.java

```diff
@@ -0,0 +1,139 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive_grid_cloud_portal.cli.cmd.sched;
+
+import static org.ow2.proactive_grid_cloud_portal.cli.CLIException.REASON_INVALID_ARGUMENTS;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.PrintStream;
+import java.util.Map;
+
+import org.apache.log4j.Logger;
+import org.ow2.proactive.scheduler.core.properties.PASchedulerProperties;
+import org.ow2.proactive.scripting.InvalidScriptException;
+import org.ow2.proactive.scripting.ScriptHandler;
+import org.ow2.proactive.scripting.ScriptResult;
+import org.ow2.proactive.scripting.SimpleScript;
+import org.ow2.proactive_grid_cloud_portal.cli.ApplicationContext;
+import org.ow2.proactive_grid_cloud_portal.cli.CLIException;
+import org.ow2.proactive_grid_cloud_portal.cli.cmd.AbstractCommand;
+import org.ow2.proactive_grid_cloud_portal.cli.cmd.Command;
+import org.ow2.proactive_grid_cloud_portal.common.SchedulerRestInterface;
+
+
+/**
+ * @author ActiveEon Team
+ * @since 13/10/2017
+ */
+public class InstallPackageCommand extends AbstractCommand implements Command {
+
+    private final String PACKAGE_PATH_NAME;
+
+    private static final String SCRIPT_PATH = "tools/InstallPackage.groovy";
+
+    private static Logger logger = Logger.getLogger(InstallPackageCommand.class);
+
+    public InstallPackageCommand(String packagePathName) {
+
+        this.PACKAGE_PATH_NAME = packagePathName;
+
+    }
+
+    @Override
+    public void execute(ApplicationContext currentContext) throws CLIException {
+
+        SchedulerRestInterface scheduler = currentContext.getRestClient().getScheduler();
+        try {
+            //retrieve the scheduler properties
+            Map<String, Object> schedulerProperties = scheduler.getSchedulerPropertiesFromSessionId(currentContext.getSessionId());
+            resultStack(currentContext).push(schedulerProperties);
+            logger.info("The scheduler properties are retrieved");
+
+            // Retrieve the script path
+            if (!validatePackagePath()) {
+                throw new CLIException(REASON_INVALID_ARGUMENTS,
+                                       String.format("'%s'does not exist or is not a valid Package path.",
+                                                     PACKAGE_PATH_NAME));
+            }
+            // Scripts binding
+            schedulerProperties.put("package.path.name", PACKAGE_PATH_NAME);
+            schedulerProperties.put("session.id", currentContext.getSessionId());
+            ScriptHandler scriptHandler = new ScriptHandler();
```

Comment:
This doesn't seem to be testable. Can you put this into a new class or private method to make this code testable